diff --git a/runtimes/msvc/BUILD.bazel b/runtimes/msvc/BUILD.bazel
--- a/runtimes/msvc/BUILD.bazel
+++ b/runtimes/msvc/BUILD.bazel
@@ -6,6 +6,7 @@
 
 _MSVC_DYNAMIC_RUNTIME_LIBRARIES = [
     "iso_stdio_wide_specifiers.lib",
+    "legacy_stdio_definitions.lib",
     "msvcprt.lib",
     "msvcrt.lib",
     "oldnames.lib",
@@ -14,6 +15,7 @@
 
 _MSVC_STATIC_RUNTIME_LIBRARIES = [
     "iso_stdio_wide_specifiers.lib",
+    "legacy_stdio_definitions.lib",
     "libcmt.lib",
     "libcpmt.lib",
     "libvcruntime.lib",
