diff --git a/toolchain/args/windows/msvc/BUILD.bazel b/toolchain/args/windows/msvc/BUILD.bazel
--- a/toolchain/args/windows/msvc/BUILD.bazel
+++ b/toolchain/args/windows/msvc/BUILD.bazel
@@ -40,13 +40,12 @@
 )
 
 cc_args(
-    name = "msvc_crt_compile_defines",
+    name = "common_crt_compile_defines",
     actions = _FRONTEND_ACTIONS,
     args = [
         "/D_CRT_NONSTDC_NO_WARNINGS",
         "/D_CRT_SECURE_NO_DEPRECATE",
         "/D_CRT_SECURE_NO_WARNINGS",
-        "/D_CRT_STDIO_ISO_WIDE_SPECIFIERS",
     ],
 )
 
@@ -252,3 +251,11 @@
     }),
     visibility = ["//visibility:public"],
 )
+
+cc_args(
+    name = "cpp_iso_stdio",
+    actions = ["@rules_cc//cc/toolchains/actions:cpp_compile", "@rules_cc//cc/toolchains/actions:cpp_header_parsing", "@rules_cc//cc/toolchains/actions:linkstamp_compile"],
+    args = ["/D_CRT_STDIO_ISO_WIDE_SPECIFIERS"],
+)
+
+cc_args_list(name = "msvc_crt_compile_defines", args = [":common_crt_compile_defines", ":cpp_iso_stdio"])
