--- a/runtimes/compiler-rt/BUILD.bazel
+++ b/runtimes/compiler-rt/BUILD.bazel
@@ -1,3 +1,9 @@
+load("@bazel_lib//lib:copy_file.bzl", "copy_file")
+load("@bazel_lib//lib:copy_to_directory.bzl", "copy_to_directory")
+
+copy_file(name = "repro_builtins_archive", src = ":clang_rt.builtins.static", out = "libclang_rt.repro_builtins.a")
+copy_to_directory(name = "repro_builtins_dir", srcs = [":repro_builtins_archive"], visibility = ["//visibility:public"])
+
 alias(
     name = "clang_rt.fuzzer.static",
     actual = "@llvm-project//compiler-rt:clang_rt.fuzzer.static",
--- a/toolchain/args/macos/BUILD.bazel
+++ b/toolchain/args/macos/BUILD.bazel
@@ -118,12 +118,13 @@
     # -l: is not supported by lld64
     args = [
         "-lSystem",
-        "{libclang_rt.builtins.a}",
+        "-L{builtins_directory}",
+        "-lclang_rt.repro_builtins",
     ],
     data = [
-        "//runtimes/compiler-rt:clang_rt.builtins.static",
+        "//runtimes/compiler-rt:repro_builtins_dir",
     ],
     format = {
-        "libclang_rt.builtins.a": "//runtimes/compiler-rt:clang_rt.builtins.static",
+        "builtins_directory": "//runtimes/compiler-rt:repro_builtins_dir",
     },
 )
